src/cool_seq_tool/mappers/exon_genomic_coords.py

```diff
@@ -754,188 +747,163 @@ async def _genomic_to_tx_segment(
-                return GenomicTxSeg(errors=[err_msg])
+        if transcript:
+            transcript_validation = await self.uta_db.validate_transcript(transcript)
+            if not transcript_validation:
+                return GenomicTxSeg(errors=[f"{transcript} does not exist in UTA"])
 
-            if not transcript:
-                # Select a transcript if not provided
-                mane_transcripts = self.mane_transcript_mappings.get_gene_mane_data(
-                    gene
-                )
+        if genomic_ac:
+            genomic_ac_validation = await self.uta_db.validate_genomic_ac(genomic_ac)
+            if not genomic_ac_validation:
+                return GenomicTxSeg(errors=[f"{genomic_ac} does not exist in UTA"])
 
-                if mane_transcripts:
-                    transcript = mane_transcripts[0]["RefSeq_nuc"]
-                else:
-                    # Attempt to find a coding transcript if a MANE transcript
-                    # cannot be found
-                    results = await self.uta_db.get_transcripts(
-                        gene=gene, alt_ac=genomic_ac
-                    )
-
-                    if not results.is_empty():
-                        transcript = results[0]["tx_ac"][0]
-                    else:
-                        # Run if gene is for a noncoding transcript
-                        query = f"""
-                            SELECT DISTINCT tx_ac
-                            FROM {self.uta_db.schema}.tx_exon_aln_v
-                            WHERE hgnc = '{gene}'
-                            AND alt_ac = '{genomic_ac}'
-                            """  # noqa: S608
-                        result = await self.uta_db.execute_query(query)
-
-                        if result:
-                            transcript = result[0]["tx_ac"]
-                        else:
-                            return GenomicTxSeg(
-                                errors=[
-                                    f"Could not find a transcript for {gene} on {genomic_ac}"
-                                ]
-                            )
-
-            tx_exons = await self._get_all_exon_coords(
-                tx_ac=transcript, genomic_ac=genomic_ac
-            )
-            if not tx_exons:
-                return GenomicTxSeg(errors=[f"No exons found given {transcript}"])
-
-            strand = Strand(tx_exons[0].alt_strand)
-            params["strand"] = strand
-
-            # Check if breakpoint occurs on an exon.
-            # If not, determine the adjacent exon given the selected transcript
-            if not self._is_exonic_breakpoint(genomic_pos, tx_exons):
-                exon_num = self._get_adjacent_exon(
-                    tx_exons_genomic_coords=tx_exons,
-                    strand=strand,
-                    start=genomic_pos if is_seg_start else None,
-                    end=genomic_pos if not is_seg_start else None,
-                )
+        if not genomic_ac:
```

Comment:
Will need to remove the new line above
```suggestion
        else:
```

Reply:
@jarbesfeld this wasn't updated in this branch